pkg/apis/crd.nsx.vmware.com/v1alpha1/ipblocksinfo_types.go

```diff
@@ -0,0 +1,42 @@
+/* Copyright © 2024 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:resource:scope="Cluster"
+
+// IPBlocksInfo is the Schema for the ipblocksinfo API
+type IPBlocksInfo struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	ExternalIPBlocks   []IPBlockInfo `json:"externalIPBlocks,omitempty"`
+	PrivateTGWIPBlocks []IPBlockInfo `json:"privateTGWIPBlocks,omitempty"`
+}
+
+type IPBlockInfo struct {
+	// Visibility specifies the visibility or scope of IPBlocks and expected
+	// consumption of IPBlocks with same scope. Can be External, Private or PrivateTGW.
+	// +kubebuilder:validation:Enum=External;Private;PrivateTGW
+	Visibility IPAddressVisibility `json:"visibility"`
```

Comment:
This one can be removed? If we will remove IPBlock ID, then we can even remove the IPBlockInfo struct, but just use slices of CIDR strings in IPBlocksInfo.

Reply:
Removed IPBlockInfo struct, thanks!